src/main/java/com/googleinterns/zoomtube/servlets/TranscriptServlet.java

```diff
@@ -37,42 +53,132 @@
     try {
-      documentBuilder = documentBuilderFactory.newDocumentBuilder();
+      DocumentBuilder documentBuilder = DocumentBuilderFactory.newInstance().newDocumentBuilder();
       document = documentBuilder.parse(new URL(transcriptXMLUrl).openStream());
       document.getDocumentElement().normalize();
     } catch (ParserConfigurationException | SAXException e) {
       // TODO: Alert the user.
       System.out.println("XML parsing error");
-      return;
+      return Optional.empty();
     }
-    NodeList nodeList = document.getElementsByTagName(TEXT_TAG);
+    return Optional.of(document);
+  }
+
+  /**
+   * Puts each transcript line from {@code document} in datastore as its own entity.
+   *
+   * @param request Indicates the lecture key to group the transcript lines under.
+   */
+  private void putTranscriptLinesInDatastore(HttpServletRequest request, Document document) {
+    long lectureId = Long.parseLong(request.getParameter(PARAM_LECTURE_ID));
+    NodeList nodeList = document.getElementsByTagName(TAG_TEXT);
     for (int nodeIndex = 0; nodeIndex < nodeList.getLength(); nodeIndex++) {
       Node node = nodeList.item(nodeIndex);
-      Element element = (Element) node;
-      String lineStart = element.getAttribute(START_ATTRIBUTE);
-      String lineDuration = element.getAttribute(DURATION_ATTRIBUTE);
-      String lineContent = node.getTextContent();
-      // TODO: Remove print statement. It is currently here for display purposes.
-      System.out.println(lineStart + " " + lineDuration + " "
-          + " " + lineContent);
+      datastore.put(createLineEntity(node, lectureId));
+    }
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    PreparedQuery preparedQuery = getLectureTranscriptQuery(request);
+    ImmutableList<TranscriptLine> transcriptLines = getTranscriptLines(preparedQuery);
+    writeResponseAsJson(response, transcriptLines);
+  }
+
+  /**
+   * Returns the query for the lecture transcripts based on lecture id indicated in {@code request}.
+   */
+  private PreparedQuery getLectureTranscriptQuery(HttpServletRequest request) {
+    long lectureId = Long.parseLong(request.getParameter(PARAM_LECTURE_ID));
+    Key lecture = KeyFactory.createKey(PARAM_LECTURE, lectureId);
+    Filter lectureFilter =
+        new FilterPredicate(TranscriptLine.PROP_LECTURE, FilterOperator.EQUAL, lecture);
+
+    Query query = new Query(TranscriptLine.ENTITY_KIND)
+                      .setFilter(lectureFilter)
+                      .addSort(TranscriptLine.PROP_START, SortDirection.ASCENDING);
+    return datastore.prepare(query);
+  }
+
+  private ImmutableList<TranscriptLine> getTranscriptLines(PreparedQuery preparedQuery) {
+    ImmutableList.Builder<TranscriptLine> lineBuilder = new ImmutableList.Builder<>();
+    for (Entity entity : preparedQuery.asQueryResultIterable()) {
+      lineBuilder.add(TranscriptLine.fromLineEntity(entity));
```

Comment:
Follow up from my suggestion builders, if this function is only used here make it a private function here, if not move it to a utils class.

Reply:
I'm a bit confused by this comment. Except for doGet and doPost, all of the methods in this class are already private. In that case, would I leave it as is?

Reply:
The earlier comment covers this, we simply shouldn't use the static method from the @AutoValue object. We be using the builder directly from the object.
When you implement #93 this should go away.

Reply:
Okay, got it. Thanks!